api/src/services/export-services/export-utils.ts

```diff
@@ -62,12 +74,88 @@ export function getQueryParams(query: SQLStatement | Knex.QueryBuilder): { text:
  * @export
  * @return {*}
  */
-export function getJsonStringifyTransformStream(): Transform {
+export function getStreamCsvTransformStream(header: string, collectionCategories?: string[]): Transform {
+  let headerStreamed = false;
   const transformStream = new Transform({
     objectMode: true, // Expects objects
     transform(chunk, _encoding, callback) {
-      // Stringify the chunk and push it to the next stream
-      callback(null, JSON.stringify(chunk));
+      if (header && !headerStreamed) {
+        // This block is executed only once
+        // Push the headers into stream
+        this.push([header, ...(collectionCategories ?? [])].join(',') + '\r\n');
+        headerStreamed = true;
+      }
+      // process chunk and push it to the next stream
+      let addLine = '\r\n';
+      if (!chunk) {
+        addLine = '';
+      }
+      callback(null, chunk + addLine);
+    }
+  });
+
+  return transformStream;
+}
+
+/**
+ * Get a query data record transform stream, that expects objects and outputs csv.
+ *
+ * Note: The incoming data stream must yield objects, or this will throw an error.
+ *
+ * @export
+ * @returns {Transform}
+ */
+export function getCsvTransformStream(
+  transformFunction: TransformFunction,
+  header: string,
+  measurementsMap?: Map<string, string>
+): Transform {
+  // Function to get either option_label or measurement_name by id
+  const getLabelById = (id: string): string | undefined => {
+    return measurementsMap ? measurementsMap.get(id) : undefined;
+  };
+
+  const processMeasData = (measData: MeasDataItem[] | undefined): void => {
+    if (measData) {
```

Comment:
Nit: rather than checking if measData is defined here, we could check it on line 155, and only call this function if it exists. And then we could remove the ` | undefined` from the parameter type.

```typescript
if(chunk.meas_data) {
   processMeasData(chunk.meas_data);
}
```